TakingITGlobal/commit2act
# [TIG-811] Mock actions, items, and add API request hooks

## Ticket

* [TIG-811](https://app.clickup.com/t/9009201449/TIG-811)

## Description

This PR adds mocks for the actions and action items from the live API data. It also updates the path param naming to use a more accurate `actionId` and `userActionId` (when an ID has been saved to a user). From there, hooks for `useGetAllActions` and `useGetItemsForAction` API requests were added, which fetch, transform, and return live data. 

**Note:** These hooks do not currently contain the translation method(s) used in the existing Log Action requests. You can find that [tracked in a separate task](https://app.clickup.com/t/9009201449/TIG-823). 

## Testing

1. Pull down and install the branch (`npm install`)
2. Run tests (`npm run test:basic`)
3. Run the local development app (`npm run start:dev`)
4. Login and use any "Log Action" prompt/link to visit the placeholder page
5. Note that both action ID and user action ID are unset, but that 6 actions are available
6. Select one of the "Enter Details" links within the page
7. Confirm the action ID is updated to match and action items display a number
8. Toggle through the links, confirming the values are updated accordingly
9. Re-click the same "Enter Details" link or back to the "Select Type" link and confirm the action ID and number of action items are not cleared and re-loaded
10. Check the browser console for logged values reflecting what you see within the page
11. Confirm the actions and items logged in console reflect the transformed versions, using camelCase naming 

## Screenshots

**Initial page with no action id set:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/ca2ca6b0-99ab-4e1b-9bf6-84e853c0020f)

**Action 5 selected with updated values rendered on the page:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/ac24b255-771e-4c5c-9280-f7b8a7c1ca4e)

**Results for URL params, actions, and action items logged in the console:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/85378854-3978-4ec6-9de3-ef5656b91987)
